app/index.js

```diff
@@ -60,10 +60,15 @@ function ConnectDialog () {
   self.channelName = ko.observable('')
   self.joinOnly = ko.observable(false)
   self.visible = ko.observable(true)
+  self.loading = ko.observable(false)
   self.show = self.visible.bind(self.visible, true)
   self.hide = self.visible.bind(self.visible, false)
   self.connect = function () {
-    self.hide()
+    self.loading(true)
```

Comment:
This should be moved into the `ui.connect` function, so it's in the same place as the call setting it back to false.